client/src/sections/resources/IndustrialFireExtinguisher.tsx

```diff
@@ -0,0 +1,97 @@
+import Line from "../../components/Line";
+import Tile from "../../components/Tile";
+import useHave from "../../hooks/useHave";
+import useGet from "../../hooks/useGet";
+import { Step, useQuestStep } from "../../hooks/useQuest";
+import { $item } from "../../util/makeValue";
+
+const IndustrialFireExtinguisher = () => {
+    const foam = useGet("_fireExtinguisherCharge");
+
+    const knobUnfinished = useQuestStep("questL05Goblin") < Step.FINISHED;
+    const knobUnused = !useGet("fireExtinguisherHaremUsed");
+    const hasPants = useHave($item`Knob Goblin harem pants`);
+    const hasVeil = useHave($item`Knob Goblin harem veil`);
+    const canVisitTheThrone = hasPants && hasVeil;
+    const showKnob =
+        useQuestStep("questL05Goblin") > Step.UNSTARTED &&
+        !canVisitTheThrone &&
+        knobUnused &&
+        knobUnfinished;
+
+    const bridgeUnfinished = useQuestStep("questL09Topping") < 1;
+    const blechUnused = !useGet("fireExtinguisherChasmUsed");
+    const showBlech =
+        useQuestStep("questL09Topping") > Step.UNSTARTED &&
+        bridgeUnfinished &&
+        blechUnused;
+
+    const batwallsUnfinished = useQuestStep("questL04Bat") < 3;
+    const batUnused = !useGet("fireExtinguisherBatHoleUsed");
+    const showBat =
+        useQuestStep("questL04Bat") > Step.UNSTARTED &&
+        batwallsUnfinished &&
+        batUnused;
+
+    const bonerUnready = [
+        useGet("cyrptAlcoveEvilness"),
+        useGet("cyrptCrannyEvilness"),
+        useGet("cyrptNicheEvilness"),
+        useGet("cyrptNookEvilness"),
+    ].some((evilness) => evilness > 25);
+    const cyrptUnused = !useGet("fireExtinguisherCyrptUsed");
+    const showCyrpt =
+        useQuestStep("questL07Cyrptic") > Step.UNSTARTED &&
+        bonerUnready &&
+        cyrptUnused;
+
+    const desertIncomplete = useGet("desertExploration") < 100;
+    const desertUnused = !useGet("fireExtinguisherDesertUsed");
+    const showDesert =
+        useQuestStep("questL11Desert") > Step.UNSTARTED &&
+        desertIncomplete &&
+        desertUnused;
+
+    if (!useHave($item`industrial fire extinguisher`) || foam <= 0)
+        return <></>;
+
+    return (
+        <Tile
+            header='Industrial Fire Extinguisher'
+            imageUrl='/images/itemimages/exting2.gif'
+        >
+            <Line>
+                Remaining foam: {foam}. That's a total of {Math.min(foam / 10)}{" "}
```

Comment:
i would recommend either this being two lines, or (my preferred solution) make the 2nd bit "(# Polar Vortex)". i don't think the sentence really adds anything but cruft here. my recommended format for this top line would be: `Foam: 80/100 (8 Polar Vortex)`. i would also add a comment noting that once we have a good tooltip structure we should tooltip polar vortex with suggested usage scenarios.